src/statement_generator.cpp

```diff
@@ -109,6 +145,76 @@ unique_ptr<CreateStatement> StatementGenerator::GenerateCreate() {
 	return create;
 }
 
+unique_ptr<AttachStatement> StatementGenerator::GenerateAttach() {
+	auto attach = make_uniq<AttachStatement>();
+	attach->info = GenerateAttachInfo();
+	return attach;
+}
+
+unique_ptr<DetachStatement> StatementGenerator::GenerateDetach() {
+	auto detach = make_uniq<DetachStatement>();
+	detach->info = GenerateDetachInfo();
+	return detach;
+}
+
+// generate USE statement
+unique_ptr<SetStatement> StatementGenerator::GenerateSet() {
+	auto name_expr = make_uniq<ConstantExpression>(GenerateDataBaseName());
+	if (RandomPercentage(90)) {
+		auto name = GetRandomAttachedDataBase();
+		name_expr = make_uniq<ConstantExpression>(Value(name));
+	}
+	auto set = make_uniq<SetVariableStatement>("schema", std::move(name_expr), SetScope::AUTOMATIC);
+	return set;
+}
+
+unique_ptr<MultiStatement> StatementGenerator::GenerateAttachUse() {
+	auto multi_statement = make_uniq<MultiStatement>();
+	multi_statement->statements.push_back(std::move(GenerateAttach()));
+	multi_statement->statements.push_back(std::move(GenerateSet()));
+	return multi_statement;
+}
+
+//===--------------------------------------------------------------------===//
+// Generate Detach Info
+//===--------------------------------------------------------------------===//
+
+unique_ptr<DetachInfo> StatementGenerator::GenerateDetachInfo() {
+	auto info = make_uniq<DetachInfo>();
+	if (RandomPercentage(20)) {
+		info->name = "RANDOM_NAME_" + RandomString(15);
+	} else {
+		info->name = GetRandomAttachedDataBase();
+	}
+	return info;
+}
+
+std::string StatementGenerator::GetRandomAttachedDataBase() {
+	auto state = GetDatabaseState(context);
+	auto st_name = state->attached_databases[RandomValue(state->attached_databases.size())];
+	auto name = st_name.get().name;
+	return name;
+}
+
+//===--------------------------------------------------------------------===//
+// Generate Attach Info
+//===--------------------------------------------------------------------===//
+
+unique_ptr<AttachInfo> StatementGenerator::GenerateAttachInfo() {
+	auto info = make_uniq<AttachInfo>();
+	auto fs = FileSystem::CreateLocal();
+	// check if the directory exists
+	if (!fs->DirectoryExists(string("../") + TESTING_DIRECTORY_NAME)) {
```

Comment:
remove the "../" these cause the testing directory to end up in a weird place

Reply:
Removed - now it puts files generated during testing to the proper `duckdb_unittest_tempdir`